### PR TITLE
Next/20200724/v8

### PR DESCRIPTION
**Pull request by victorjulien**
#5212 #5210 
gitignore update
unittest cleanups
ssh hassh logging change to allow ES to accept it

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
Passed

suricata-verify-pr: 280

### CODE REVIEW

File: rust/src/ssh/logger.rs [hunk 1]
```diff
@@ -28,11 +28,15 @@ fn log_ssh(tx: &SSHTransaction, js: &mut JsonBuilder) -> Result<bool, JsonError>
         if tx.cli_hdr.swver.len() > 0 {
             js.set_string_from_bytes("software_version", &tx.cli_hdr.swver)?;
         }
-        if tx.cli_hdr.hassh.len() > 0 {
-            js.set_string_from_bytes("hassh", &tx.cli_hdr.hassh)?;
-        }
-        if tx.cli_hdr.hassh_string.len() > 0 {
-            js.set_string_from_bytes("hassh.string", &tx.cli_hdr.hassh_string)?;
+        if tx.cli_hdr.hassh.len() > 0 || tx.cli_hdr.hassh_string.len() > 0 {
+            js.open_object("hassh")?;
```
Comment (victorjulien) [inline rust/src/ssh/logger.rs:32]: Avoid type colision by adding this `hassh` object with `hash` and `string` members like with `ja3` and `ja3s`.

Comment (catenacyber) [reply rust/src/ssh/logger.rs:32]: What do you mean ?
I do not see the bug...

**Comment by victorjulien** [comment]
> What do you mean ?
I do not see the bug... 

This is what I got with ES (through evebox):
```
2020-07-22 14:07:54  ERROR evebox::elastic::importer: Elasticsearch one of more errors to the commit operation, first error: mapper [ssh.client.hassh] of different type, current_type [text], merged_type [ObjectMapper]
```
Pretty much `ssh.client.hassh` was the MD5 String, but then `ssh.client.hassh.string` was the string. So here `ssh.client.hassh` was an intermediate object. ES refused it because it can't be both.

**Comment by catenacyber** [comment]
Thanks for the explanation 👍

**Pull request merged**